### PR TITLE
Feature/237 frontend roles

### LINKED ISSUE #237: Add helper methods for roles in frontend
**Is your feature request related to a problem? Please describe.**
We currently have no way to check for permissions on the frontend and to translate a user's role number into text.

**DoD**
- Helper methods to check if the currentUser has 'clearance' of a specific role.
- Method to translate the role number into text for display.

### PR DESCRIPTION
**Pull request by TE4-jacob-lundberg**
## What issue did you implement or fix?
Fixes #237 

## Summary
- Fixed all linting errors in UsersModule
- Added method for translating the role number into their highest clearance.
- Added helper class to use when checking permissions w/ example in findview
- Added testsfile/tests for UsersModule

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes

## How can this implementation be improved?
- The static methods do contain quite a bit of duplication, I'm open to suggestions on how to improve it.

### CODE REVIEW

File: frontend/src/helpers/roleChecker.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  }
+
+  static studentHealth(): boolean {
+    return UsersModule.currentUser && (UsersModule.currentUser.role & 2) == 2 || (UsersModule.currentUser.role > 2);
+  }
+
+  static student(): boolean {
+    return UsersModule.currentUser && (UsersModule.currentUser.role & 1) == 1;
+  }
+}
```
Comment (TE4-Mattis-Abrahamsson) [inline frontend/src/helpers/roleChecker.ts]: Correct me if I'm wrong but JS class files should be named with PascalCase, right?

Comment (TE4-jacob-lundberg) [reply frontend/src/helpers/roleChecker.ts]: You are correct, will update

Comment (TE4-Mattis-Abrahamsson) [inline frontend/src/helpers/roleChecker.ts]: Maybe the functions should be called isAdmin and isPrincipal etc. to make it clear that they return a boolean? I think this is also a JS standard.

Comment (TE4-Mattis-Abrahamsson) [reply frontend/src/helpers/roleChecker.ts]: 👍

Comment (TE4-jacob-lundberg) [reply frontend/src/helpers/roleChecker.ts]: once again, you are correct. Would want a '?' even but js does not allow that in function names

Comment (TE4-Mattis-Abrahamsson) [reply frontend/src/helpers/roleChecker.ts]: Yup, it's too bad

File: frontend/src/helpers/RoleChecker.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  }
+
+  static isStudentHealth(): boolean {
+    return UsersModule.currentUser && (UsersModule.currentUser.role & 2) == 2 || (UsersModule.currentUser.role > 2);
+  }
+
+  static isStudent(): boolean {
+    return UsersModule.currentUser && (UsersModule.currentUser.role & 1) == 1;
+  }
+}
```
Comment (TE4-Mattis-Abrahamsson) [inline frontend/src/helpers/RoleChecker.ts]: We could break it out to a helper function which checks if currentUser is set and if role matches the given number

File: frontend/src/helpers/RoleChecker.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import UsersModule from '../store/modules/UsersModule';
+
+export default class RoleChecker {
+  
+  static isPrincipal(): boolean {
+    return UsersModule.currentUser && (UsersModule.currentUser.role & 32) == 32 || (UsersModule.currentUser.role > 32);
```
Comment (TE4-Daniel-Kull) [inline frontend/src/helpers/RoleChecker.ts]: Just to avoid confusion you might want to add an extra parenthesis around the OR statement. Currently, it's not clear which part it corresponds to.

File: frontend/src/helpers/RoleChecker.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  }
+
+  static isStudentHealth(): boolean {
+    return this.checkPermission(UsersModule.currentUser, 2);
+  }
+
+  static isStudent(): boolean {
+    return UsersModule.currentUser && (UsersModule.currentUser.role & 1) == 1;
+  }
+}
```
Comment (TE4-Mattis-Abrahamsson) [inline frontend/src/helpers/RoleChecker.ts]: As user is always the same in checkPermissionsit might not need to be passed as an argument?

File: frontend/src/store/modules/UsersModule.ts [hunk 5]
```diff
@@ -42,45 +43,59 @@ class UsersModule extends VuexModule {
...
+  }
+
   @Action({rawError: true})
   public fetchAll(): Promise<object> {
     return new Promise((resolve, reject) => {
       UsersAPI.all()
         .then((response: User[]) => {
           this.convertUserList(response);
-          resolve();
+          resolve(this._users);
```
Comment (TE4-Mattis-Abrahamsson) [inline frontend/src/store/modules/UsersModule.ts]: Why was this added?

File: frontend/src/store/modules/UsersModule.ts [hunk 6]
```diff
@@ -9,26 +9,27 @@ import {
...
 
 export interface UserCollection { [uid: number]: User; }
 
 @Module({dynamic: true, namespaced: true, name: 'UsersModule', store})
 class UsersModule extends VuexModule {
-  private _failure: any = null;
+  private _failure: object = {};
   public _users: UserCollection = {};
-  private _currentUser: any = null;
+  private _currentUser: number = 0;
```
Comment (TE4-Mattis-Abrahamsson) [inline frontend/src/store/modules/UsersModule.ts]: Won't this possibly break something, as the getter will now return 0 when no user is set?

**Comment by TE4-oskar-pilborg** [comment]
W8 with merge

File: frontend/src/store/modules/UsersModule.ts (hunk 6 above)
Comment (TE4-jacob-lundberg) [reply frontend/src/store/modules/UsersModule.ts]: It shouldn't as all possible id's contain multiple different numbers based on their google_token

File: frontend/src/store/modules/UsersModule.ts (hunk 5 above)
Comment (TE4-jacob-lundberg) [reply frontend/src/store/modules/UsersModule.ts]: I believe in order to add more consistency. Most other methods did return the data with their promise so I added it here is as well.

File: frontend/src/helpers/RoleChecker.ts (hunk 4 above)
Comment (TE4-jacob-lundberg) [reply frontend/src/helpers/RoleChecker.ts]: Right, will change it for now. Wanted it to take a user as an argument in case of future usage but should probably wait with it.

File: frontend/src/store/modules/UsersModule.ts (hunk 5 above)
Comment (TE4-Mattis-Abrahamsson) [reply frontend/src/store/modules/UsersModule.ts]: Right, however, the data could never really be used as it won't be reactive hence why we have a getter for users, perhaps it could return a success message instead?

File: frontend/src/store/modules/UsersModule.ts (hunk 6 above)
Comment (TE4-Mattis-Abrahamsson) [reply frontend/src/store/modules/UsersModule.ts]: so if there is an if() statement on currentUserID it will still be false?

File: frontend/src/helpers/RoleChecker.ts [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  static isTeacher(): boolean {
+    return this.checkPermission(UsersModule.currentUser, 4);
+  }
+
+  static isStudentHealth(): boolean {
+    return this.checkPermission(UsersModule.currentUser, 2);
+  }
+
+  static isStudent(): boolean {
+    return UsersModule.currentUser && (UsersModule.currentUser.role & 1) == 1;
```
Comment (TE4-Mattis-Abrahamsson) [inline frontend/src/helpers/RoleChecker.ts]: Also, is this a special case or why can it not use the helper function?

Comment (TE4-jacob-lundberg) [reply frontend/src/helpers/RoleChecker.ts]: Special case

File: frontend/src/store/modules/UsersModule.ts (hunk 6 above)
Comment (TE4-jacob-lundberg) [reply frontend/src/store/modules/UsersModule.ts]: Falsey so yes
